lib/tlIO/FFmpegWrite.cpp

```diff
@@ -2,32 +2,150 @@
 // Copyright (c) 2021-2023 Darby Johnston
 // All rights reserved.
 
-#include <tlIO/FFmpeg.h>
-
 #include <tlCore/StringFormat.h>
+#include <tlCore/AudioResample.h>
+#include <tlCore/LogSystem.h>
+
+#include <tlIO/FFmpeg.h>
 
 extern "C"
 {
-#include <libavcodec/avcodec.h>
+    
+#include <libavutil/audio_fifo.h>
 #include <libavutil/imgutils.h>
 #include <libavutil/opt.h>
+#include <libavutil/timestamp.h>
+    
+#include <libavcodec/avcodec.h>
+    
 }
 
 namespace tl
 {
     namespace ffmpeg
     {
+        namespace
+        {
+            
+            AVSampleFormat toPlanarFormat(const enum AVSampleFormat s)
+            {
+                switch(s)
+                {
+                case AV_SAMPLE_FMT_U8:
+                    return AV_SAMPLE_FMT_U8P;
+                case AV_SAMPLE_FMT_S16:
+                    return AV_SAMPLE_FMT_S16P;
+                case AV_SAMPLE_FMT_S32:
+                    return AV_SAMPLE_FMT_S32P;
+                case AV_SAMPLE_FMT_FLT:
+                    return AV_SAMPLE_FMT_FLTP;
+                case AV_SAMPLE_FMT_DBL:
+                    return AV_SAMPLE_FMT_DBLP;
+                default:
+                    return s;
+                }
+            }
+            
+            //! Check that a given sample format is supported by the encoder
+            bool checkSampleFormat(
+                const AVCodec* codec, enum AVSampleFormat sample_fmt)
+            {
+                const enum AVSampleFormat *p = codec->sample_fmts;
+
+                while (*p != AV_SAMPLE_FMT_NONE) {
```

Comment:
Please format the code consistently like the other code in the file.

Reply:
Hopefully fixed.

Reply:
For example:
* Always initialize variables when they are declared, and declare just one variable per line:
```
const AVChannelLayout* p = nullptr;
const AVChannelLayout* best_ch_layout = nullptr;
```
* Try and have a single return statement in functions.
* Brackets are on their own lines:
```
if (...)
{
    ...
}
```